pkg/client/unversioned/helper.go

```diff
@@ -577,3 +607,26 @@ func TimeoutFromListOptions(options unversioned.ListOptions) time.Duration {
 	}
 	return 0
 }
+
+func stripAPIFromPrefix(base string) string {
```

Comment:
Considering the pain I went through to make the proxy command in kubectl work, I think any solution for which this function is required is going to be a real pain going forward. Instead I suggest:

* Deprecate / start ignoring the "prefix" field in our config file. That was just a mistake to have at all, given the current implementation.
* Add a new field. I'm not sure what to call it (parent-prefix? proxy-prefix?). But it should be blank by default, and if set, it should get prepended to everything, even AbsPath calls.

I think if you do this, this PR will be a lot smaller and simpler.